app/models/UserModel.js

```diff
@@ -0,0 +1,58 @@
+const mongoose = require('mongoose');
+var crypto = require('crypto');
+var strava = require('strava-v3');
+
+// Crypto algorithm.
+var algorithm = 'aes-256-ctr';
+var cryptoPassword = process.env.KILOMETRIKISA_CRYPTO_PASSWORD;
+
+// Define schema for user.
+var UserSchema = new mongoose.Schema({
+  stravaUserId: { type: Number },
+  stravaToken: { type: String },
+  tokenExpire: { type: Number },
+  refreshToken: { type: String },
+  kilometrikisaToken: { type: String },
+  kilometrikisaSessionId: { type: String },
+  kilometrikisaUsername: { type: String },
+  kilometrikisaPassword: { type: String },
+  email: { type: String },
+
+  // Sync kilometers automatically.
+  autosync: { type: Boolean },
+
+  // If true, we know that notification email about session timeout
+  // is sent to user.
+  notifiedByEmail: { type: Boolean },
+});
+
+UserSchema.methods.updateToken = function() {
+  var d = new Date();
+  if (d > this.tokenExpire) {
+    const payload = strava.oauth.refreshToken(this.refreshToken);
+    this.stravaToken = payload.access_token;
+    this.tokenExpire = payload.expires_at * 1000;
+    this.refreshToken = payload.refresh_token;
```

Comment:
This should be like in PR #32

Reply:
Thanks for the pointer and PR @Mattti0 ! 😍 

I've now merged #32 into this PR.